### PR TITLE
Change how we're reseting answers

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Reset values for dependant derived questions
Only reset question values if the selected answer is not present in the answer options

### CODE REVIEW

File: app/models/case_log.rb [hunk 1]
```diff
@@ -213,12 +213,33 @@ def update_status!
...
+      enabled = form.enabled_page_questions(self)
+      contains_selected_answer_option = enabled.map(&:id).include?(question.id) && enabled.find { |q| q.id == question.id }.answer_options.values.map { |x| x["value"] }.include?(public_send(question.id))
+      unless contains_selected_answer_option
+        public_send("#{question.id}=", nil) if respond_to?(question.id.to_s)
+      end
+    end
+  end
+
+  def reset_derived_questions
+    dependent_questions = { layear: [{ key: :renewal, value: "No" }], homeless: [{ key: :renewal, value: "No" }] }
```
Comment (baarkerlounger) [inline app/models/case_log.rb:227]: Can we derive this from the JSON rather than hardcoding it?

Comment (kosiakkatrina) [reply app/models/case_log.rb:227]: We don't really have anything that indicates that these are inferred in the JSON. Would it make sense to add it just for this reason?

Comment (baarkerlounger) [reply app/models/case_log.rb:227]: hmm we have:

```
"layear": {
    "depends_on":  [{ "renewal": "No" }]
},
"homeless": {
    "depends_on":  [{ "renewal": "No" }]
}
```

If we collect those and apply this does it break for other fields?

Comment (kosiakkatrina) [reply app/models/case_log.rb:227]: So we're already getting the depends on to determine if something is not routed to and if it's not we clear it.
In this case we want to clear something that is routed to, but only if it's one of the fields we're inferring. We can get the questions that are routed to from this JSON, but we would still need to filter them down to the ones that get inferred, no?

Comment (kosiakkatrina) [reply app/models/case_log.rb:227]: Otherwise we'd just clear everything?

Comment (baarkerlounger) [reply app/models/case_log.rb:227]: Ah yeah I remember now.  :+1:

**Comment by baarkerlounger** [comment]
Think you'll need to bump nokogiri on main before you can merge?

**Pull request merged**